src/libraries/System.Private.CoreLib/src/System/AppContext.cs

```diff
@@ -135,6 +135,23 @@ public static void SetSwitch(string switchName, bool isEnabled)
             }
         }
 
+        internal static void LogSwitchValues(RuntimeEventSource ev)
+        {
+            if (s_switches is null)
+            {
+                return;
+            }
+
+            lock (s_switches)
+            {
+                foreach (var (k, v) in s_switches)
```

Comment:
This is only going to log switches set using `AppContext.SetSwitch` . It won't log switches set using build properties and runtime.json that I think are much more common these days. Do we need to log those too?

Reply:
Yes, I would like to include those as well -- do you know which API they go through if not SetSwitch?

Reply:
`AppDomain.SetData` (https://github.com/dotnet/runtime/issues/47922 tracks adding `AppContext.SetData` for completeness)

Also, the runtime startup goes over everything that is in runtime.json and calls `AppDomain.SetData` for it.

Reply:
In that case I will also add everything in `s_dataStore` which has a boolean value. I'm trying to avoid grabbing everything in s_dataStore to limit the amount of potential user data that's recorded.